### PR TITLE
{data} [intel/2017b] bat 0.3.3 (REVIEW)

### PR DESCRIPTION
**Pull request by wpoely86**
Needs:
- ~~easybuilders/easybuild-framework#2353~~
- ~~easybuilders/easybuild-easyblocks#1299~~

### CODE REVIEW

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 2 out of 4 (4 easyconfigs in this PR)
nic200 - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E7-8891 v4 @ 2.80GHz, Python 2.7.5
See https://gist.github.com/410459f1466ec74c33895fb423e0f994 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
nic200 - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E7-8891 v4 @ 2.80GHz, Python 2.7.5
See https://gist.github.com/dd95de78c52b278da36d21ba3614e31e for a full test report.

File: easybuild/easyconfigs/a/arrow/arrow-0.7.1-intel-2017b-Python-3.6.3.eb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+version = '0.7.1'
+versionsuffix = '-Python-%(pyver)s'
+
+homepage = 'https://arrow.apache.org'
+description = """Apache Arrow is a cross-language development platform for in-memory data."""
+
+toolchain = {'name': 'intel', 'version': '2017b'}
+
+sources = ['apache-arrow-%(version)s.tar.gz']
+source_urls = ['https://github.com/apache/arrow/archive/']
```
Comment (boegel) [inline easybuild/easyconfigs/a/arrow/arrow-0.7.1-intel-2017b-Python-3.6.3.eb]: @wpoely86 please include a SHA256 checksum

File: easybuild/easyconfigs/a/arrow/arrow-0.7.1-intel-2017b-Python-3.6.3.eb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+preconfigopts = "ZLIB_HOME=$EBROOTZLIB"
+configopts = "-DARROW_PYTHON=ON -DARROW_PLASMA=ON -DARROW_BUILD_TESTS=ON -DCMAKE_BUILD_TYPE=Release"
+
+start_dir = 'cpp'
+separate_build_dir = True
+
+builddependencies = [('CMake', '3.9.5')]
```
Comment (boegel) [inline easybuild/easyconfigs/a/arrow/arrow-0.7.1-intel-2017b-Python-3.6.3.eb]: please move up, right below `sources` block

File: easybuild/easyconfigs/a/arrow/arrow-0.7.1-intel-2017b-Python-3.6.3.eb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        'start_dir': 'python',
+        'source_tmpl': sources[0],
+        'buildcmd': 'build_ext',
+        'buildopts': '--with-plasma --inplace --build-type=release',
+    }),
+]
+
+modextrapaths = {'PYTHONPATH': ['lib/python%(pyshortver)s/site-packages']}
+
+dependencies = [
```
Comment (boegel) [inline easybuild/easyconfigs/a/arrow/arrow-0.7.1-intel-2017b-Python-3.6.3.eb]: please move up, same block as `builddependencies`

File: easybuild/easyconfigs/b/bat/bat-0.3.3-intel-2017b-Python-3.6.3.eb [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+versionsuffix = '-Python-%(pyver)s'
+
+homepage = 'https://pypi.python.org/pypi/bat'
+description = """The BAT Python package supports the processing and analysis of Bro data with Pandas,
+scikit-learn, and Spark"""
+
+toolchain = {'name': 'intel', 'version': '2017b'}
+
+# this is a bundle of Python packages
+exts_defaultclass = 'PythonPackage'
```
Comment (boegel) [inline easybuild/easyconfigs/b/bat/bat-0.3.3-intel-2017b-Python-3.6.3.eb]: please move down, right above `exts_list` (and drop the comment, it's clear without it imho)

File: easybuild/easyconfigs/b/bat/bat-0.3.3-intel-2017b-Python-3.6.3.eb [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+dependencies = [
+    ('Python', '3.6.3'),
+    ('PyYAML', '3.12', versionsuffix),
+    ('scikit-learn', '0.19.1', versionsuffix),
+    ('arrow', '0.7.1', versionsuffix),
+    ('Spark', '2.2.0', '-Hadoop-2.6-Java-1.8.0_152%s' % versionsuffix),
```
Comment (boegel) [inline easybuild/easyconfigs/b/bat/bat-0.3.3-intel-2017b-Python-3.6.3.eb]: please use `%(versionsuffix)s`

File: easybuild/easyconfigs/b/bat/bat-0.3.3-intel-2017b-Python-3.6.3.eb [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+exts_list = [
+    ('requests', '2.18.4', {
+        'source_urls': ['https://pypi.python.org/packages/source/r/requests/'],
+        'checksums': ['9c443e7324ba5b85070c4a818ade28bfabedf16ea10206da1132edaa6dda237e'],
+    }),
+    ('pathtools', '0.1.2', {
+        'source_urls': ['https://pypi.python.org/packages/source/p/pathtools'],
+        'checksums': [
+            '7c35c5421a39bb82e58018febd90e3b6e5db34c5443aaaf742b3f33d4655f1c0',  # pathtools-0.1.2.tar.gz
```
Comment (boegel) [inline easybuild/easyconfigs/b/bat/bat-0.3.3-intel-2017b-Python-3.6.3.eb]: let's collapse this to a single line (`--inject-checksums` in `develop` should do so automagically)

File: easybuild/easyconfigs/j/jemalloc/jemalloc-5.0.1-GCCcore-6.4.0.eb [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+version = '5.0.1'
+
+homepage = 'http://www.canonware.com/jemalloc'
+description = """jemalloc is a general purpose malloc(3) implementation that emphasizes fragmentation avoidance and
+ scalable concurrency support."""
+
+toolchain = {'name': 'GCCcore', 'version': '6.4.0'}
+
+source_urls = ['https://github.com/jemalloc/jemalloc/archive']
+sources = ['%(version)s.tar.gz']
```
Comment (boegel) [inline easybuild/easyconfigs/j/jemalloc/jemalloc-5.0.1-GCCcore-6.4.0.eb]: please include SHA256 checksum

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
nic200 - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E7-8891 v4 @ 2.80GHz, Python 2.7.5
See https://gist.github.com/638064c164faee0dd1102f5facbc9f9b for a full test report.

File: easybuild/easyconfigs/s/Spark/Spark-2.2.0-intel-2017b-Hadoop-2.6-Java-1.8.0_152-Python-3.6.3.eb [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+    ('pyspark', version, {
+        'source_tmpl': 'spark-%(version)s-bin-hadoop2.6.tgz',
+        'start_dir': 'python',
+        'checksums': ['86f33f6f4fe545cb848600004144f09b6e8413af47ce02c1203c3ecda075288d'],
+    }),
+]
+
+modextrapaths = {'PYTHONPATH': ['lib/python%(pyshortver)s/site-packages']}
+
+dependencies = [
```
Comment (boegel) [inline easybuild/easyconfigs/s/Spark/Spark-2.2.0-intel-2017b-Hadoop-2.6-Java-1.8.0_152-Python-3.6.3.eb]: @wpoely86 Please move this up too, missed that one earlier...

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 2 out of 4 (4 easyconfigs in this PR)
node2454.golett.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/58473fddefc005c0da64c6790ee8fd6e for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 2 out of 4 (4 easyconfigs in this PR)
node2066.delcatty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/938a2a54e9c719b362960e4b53673344 for a full test report.

**Comment by wpoely86** [comment]
@boegel you must use develop from today

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
node2454.golett.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/b11a8c1715c42e1ee762ae3939a1feea for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
node2066.delcatty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/a33473a85d2c3f5741bfb04aba4134aa for a full test report.

**Comment by boegel** [comment]
Going in, thanks @wpoely86!

**Pull request merged**